src/SimSpace/SimFormat.hs

```diff
@@ -483,12 +484,26 @@ reformat regroup programLines = do
       -> ([Either BlankLine SortedImportStmts], [Block])
       -> m [Block]
     reassemble nonImports (chunkedImports, leftovers) = do
-        rechunked <- (if regroup then rechunk else pure) chunkedImports
+        rechunked <- (if regroup then rechunk else pure) (requalify chunkedImports)
         pure $
           fmap Left nonImports
           <> (fmap . fmap) renderImportStmts rechunked
           <> leftovers
       where
+        -- If we are using post qualified imports, turn the regular qualified ones into post qualified ones
+        requalify :: [Either BlankLine SortedImportStmts] -> [Either BlankLine SortedImportStmts]
+        requalify ci
+          | any hasPostQualified (rights ci) =
+            fmap (SortedImportStmts . Map.mapKeys requalifyImportStmt . unSortedImportStmts) <$> ci
+          | otherwise = ci
+          where
+            hasPostQualified :: SortedImportStmts -> Bool
+            hasPostQualified = any importStmtHeadPostQualified . Map.keys . unSortedImportStmts
+            requalifyImportStmt :: ImportStmtHead -> ImportStmtHead
+            requalifyImportStmt stmt
+              | importStmtHeadQualified stmt = stmt {importStmtHeadQualified = False, importStmtHeadPostQualified = True}
```

Comment:
Is this enough to perform the post qualification? :nice:

Reply:
The post qualify support is already in there ;) This bit just renormalizes non post-qualified ones to post-qualified if its enabled in the module. (So we don't end up with a mix of post and non-post ones in the same file)